### PR TITLE
MM-48456 - restrict analytics call to only admins

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR makes sure that the show start trial modal only calls the analytics endpoint when the user is an admin user.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48456

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/system_notice/system_notice.tsx [hunk 1]
```diff
@@ -34,13 +34,13 @@ type Props = {
...
     componentDidMount() {
-        if (this.props.isSystemAdmin) {
+        if (this.props.isSystemAdmin && !this.props.analytics) {
             this.props.actions.getStandardAnalytics();
         }
     }
 
     componentDidUpdate(prevProps: Props) {
-        if (prevProps.isSystemAdmin !== this.props.isSystemAdmin && this.props.isSystemAdmin) {
+        if ((prevProps.isSystemAdmin !== this.props.isSystemAdmin && this.props.isSystemAdmin && !this.props.analytics)) {
```
Comment (agarciamontoro) [inline components/system_notice/system_notice.tsx:43]: Is it possible that `this.props.analytics` is outdated and we do need the call to getStandardAnalytics to update them? Or can we assume if they're available, they're updated? I have no idea, but it seemed risky.

File: components/announcement_bar/announcement_bar_controller.tsx [hunk 2]
```diff
@@ -87,6 +87,13 @@ class AnnouncementBarController extends React.PureComponent<Props> {
             );
         }
 
+        let autoStartTrialModal = null;
+        if (this.props.userIsAdmin) {
+            autoStartTrialModal = (
+                <AutoStartTrialModal/>
+            );
+        }
```
Comment (agarciamontoro) [inline components/announcement_bar/announcement_bar_controller.tsx:95]: Yay, thanks!

File: components/announcement_bar/show_start_trial_modal/show_start_trial_modal.tsx [hunk 3]
```diff
@@ -59,7 +59,7 @@ const ShowStartTrialModal = () => {
         if (!stats?.TOTAL_USERS) {
             dispatch(getStandardAnalytics());
         }
-    }, [!stats?.TOTAL_USERS]);
+    }, [stats?.TOTAL_USERS]);
```
Comment (agarciamontoro) [inline components/announcement_bar/show_start_trial_modal/show_start_trial_modal.tsx:62]: This is unrelated, right?

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Thank you, @pvev! Just a question below on something I don't completely understand.

File: components/system_notice/system_notice.tsx (hunk 1 above)
Comment (pvev) [reply components/system_notice/system_notice.tsx:43]: for my investigation, it seems that system_notice and showStartTrialModal load at same times/circumstances, so I assume it would be safe to consider the state of the analytics would be the same for both components, however now that you mention it, I am not sure if it is worth the potential risk of system_notice getting outdated analytics information vs the benefit on not calling analytics again??

File: components/announcement_bar/show_start_trial_modal/show_start_trial_modal.tsx (hunk 3 above)
Comment (pvev) [reply components/announcement_bar/show_start_trial_modal/show_start_trial_modal.tsx:62]: yes, and a merge problem, the idea was to completely remove that dependency which is unnecessary

**Comment by pvev** [comment]
to solve the immediate problem of stop non-admin users to make calls to the analytics endpoint given the fact that this is associated to a customer ticket, we can merge this PR and then continue review https://github.com/mattermost/mattermost-webapp/pull/11790 which makes sure to replace the other places where the analytics endpoint was called only for grabbing the total_users count.
@agarciamontoro @hmhealey ^^

**Review by hmhealey — APPROVED** [review]

**Comment by hmhealey** [comment]
Sorry about the delay there. I'm not sure how I missed reviewing this when I added the labels and everything to it

**Comment by pvev** [comment]
@agarciamontoro friendly reminder on this one :)

**Review by agarciamontoro — APPROVED** [review]
Ship it!

**Review by stevemudie — APPROVED** [review]
This is all set @pvev

**Comment by pvev** [comment]
> This is all set @pvev

thank you @stevemudie ❤️
